ocs/rook.py

```diff
@@ -0,0 +1,142 @@
+"""
+A module for all rook functionalities and abstractions.
+
+This module has rook related classes, support for functionalities to work with
+rook cluster. This works with assumptions that an OCP cluster is already
+functional and proper configurations are made for interaction.
+"""
+
+import logging
+
+import oc.openshift_ops as ac
+from ocs.pod import Pod
+
+logger = logging.getLogger(__name__)
+
+
+class Rook(object):
+    """
+    High level rook abstraction. This class should handle operator
+    and cluster objects.
+
+    As of now deligated rook environment to corresponding cluster object
+    but in future rookenv should be handled here.Also few operator related ops
+    will be included in this class. Keeping this class lightweight for the first
+    cut.
+
+    Attrs:
+        self._cluster (RookCluster): current cluster object
+    """
+
+    def __init__(self, **config):
+        self._cluster = RookCluster(**config)
+
+    @property
+    def cluster(self):
+        return self._cluster
+
+
+class RookCluster(object):
+    """
+    Handles all cluster related operations from ceph perspective
+
+    This class has depiction of ceph cluster. Contains references to
+    pod objects which represents ceph cluster entities.
+
+    Attributes:
+        _ocs_pods (list) : A list of  ceph cluster related pods
+
+        _cluster_name (str): Name of ceph cluster.
+
+        _namespace (str): openshift Namespace where this cluster lives.
+
+        _api_client (APIClient): api-client used for interacting with openshift.
+
+        _url_prefix (str): url prefix for REST api interaction
+
+    Methods:
```

Comment:
Where are those methods implemented? I don't think that we should have even document methods as they are documented in their own docstrings.

Reply:
Sorry, this was written for my reference but forgot to remove it. Done